review_practice/react-app/src/containers/RoomList/ReservationModalBody.js

```diff
@@ -0,0 +1,87 @@
+    const { date, personnel } = searchOptions;
+    const { dispatchSearchOption } = useContext(SearchRoomContext);
+    const [addBooking, { data, loading, error }] = useMutation(RoomAPI.ADD_BOOKING);
+
+    useEffect(() => {
+        if (data && !data.reserveRoom.success) alert('이미 예약되어있는 숙소입니다');
+        if (data && data.reserveRoom.success) {
+            closeModal();
+            alert('예약이 완료됬습니다.');
+            dispatchSearchOption({ type: 'reset' });
+        }
+        if (loading) console.log('추가 중');
+        if (error) alert('이미 예약되어있는 숙소입니다');
+    }, [data, loading, error]);
+
+    const nights = date
+        ? getNightValueFromDateString({ startDateString: date.checkIn, endDateString: date.checkOut })
+        : 0;
+    const guest = personnel ? personnel.adult + personnel.children : 0;
+
+    const reserveButtonClick = () => {
+        const variables = { data: { roomId: room.id, date, personnel } };
+        addBooking({ variables });
+    };
+
+    return (
+        <Style.ReservationModalBody>
+            {loading && <small className="Loading-Statement">예약 처리중입니다. 잠시만 기달려주세요...</small>}
+            <h2>{room.name}</h2>
+            <hr></hr>
+            <label htmlFor="date">날짜</label>
+            {!date && <input className="Text-Center" id="date" type="text" value="날짜를 입력해주세요" readOnly />}
+            {date && (
+                <Style.DateWrapper id="date">
+                    <input className="Date-Input" id="start-date" type="text" value={date.checkIn} readOnly />
+                    <span className="Date-Arrow">></span>
+                    <input className="Date-Input" id="end-date" type="text" value={date.checkOut} readOnly />
+                </Style.DateWrapper>
+            )}
+            <label htmlFor="personnel">인원</label>
+            {personnel ? (
+                <input
+                    id="personnel"
+                    type="text"
+                    value={`게스트 ${guest}명${personnel.infant === 0 ? '' : `, 유아 ${personnel.infant}명`}`}
+                    readOnly
+                />
+            ) : (
+                <input className="Text-Center" id="personnel" type="text" value="인원을 설정해주세요" readOnly />
+            )}
+            <div className="Price-Wrapper">
+                {date && personnel ? (
+                    <>
+                        <span>
+                            ￦ {room.price}원 X 게스트 {guest}명 X {nights}박
+                        </span>
+                        <span>
+                            <strong>￦ {room.price * guest * nights}원</strong>
+                        </span>
+                    </>
+                ) : (
+                    <div className="Text-Center">숙박조건을 모두 입력해주세요</div>
+                )}
+            </div>
+            <hr></hr>
+            <button
+                className={!(date && personnel) ? 'Non-Clickable' : ''}
+                onClick={!(date && personnel) ? () => {} : reserveButtonClick}
```

Comment:
```suggestion
                onClick={reserveButtonClick}
```

`disabled` 를 사용할 경우 굳이 비활성화 상태일 때 빈 핸들러를 보내주는 수고를 할 필요가 없게 됩니다.